### PR TITLE
run multiple actions via filter views

### PR DESCRIPTION
**Pull request by christianhugoch**
- show a 'action_row_var' select for actions in filter views  (not for clear)
  with none, state, eatch_matching_row
- none: nothing changes
- state: the action gets a row with the values from the state

- eatch_matching_row:
    - all rows, matching the state, are loaded and the action is called for all rows
    - paging and order is ignored, but eatch_matching_row has an optional row limit
    - when there are multiple 'reload_page, goto or popup' results then take the first
    - multiple 'notifies, errors or downloads' are returned as array (limited to 5)

### CODE REVIEW

File: packages/saltcorn-data/base-plugin/viewtemplates/filter.js [hunk 1]
```diff
@@ -495,17 +528,83 @@ const eq_string = (x, y) => `${x}` === `${y}`;
...
+  const initOrPush = (newElement, memberName) => {
+    if (result.json[memberName]) result.json[memberName].push(newElement);
+    else result.json[memberName] = [newElement];
+  };
+  const initOnce = (newElement, memberName) => {
+    if (typeof result[memberName] === "undefined")
+      result.json[memberName] = newElement;
+  };
+  for (const result of results) {
+    if (result.reload_page) initOnce(result.reload_page, "reload_page");
```
Comment (glutamate) [inline packages/saltcorn-data/base-plugin/viewtemplates/filter.js:576]: before this line insert:

`if(!result) continue;`

not all actions will return an object and then it gives error

**Review by glutamate — CHANGES_REQUESTED** [review]
see small comments

File: packages/saltcorn-data/base-plugin/viewtemplates/filter.js [hunk 2]
```diff
@@ -495,17 +528,83 @@ const eq_string = (x, y) => `${x}` === `${y}`;
...
   table_id,
   viewname,
-  { columns, layout },
+  config,
   body,
   { req, res },
   { actionQuery }
 ) => {
-  const result = await actionQuery();
+  const table = Table.findOne({ id: table_id });
```
[the remark this answers is not in the document]
Comment (christianhugoch) [reply packages/saltcorn-data/base-plugin/viewtemplates/filter.js]: changed it, and in actionQuery, too

File: packages/saltcorn-data/base-plugin/viewtemplates/filter.js (hunk 1 above)
Comment (christianhugoch) [reply packages/saltcorn-data/base-plugin/viewtemplates/filter.js:577]: fixed

**Review by glutamate — APPROVED** [review]

**Pull request merged**